pkg/flypg/http.go

```diff
@@ -0,0 +1,120 @@
+package flypg
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+
+	"github.com/superfly/flyctl/pkg/agent"
+)
+
+type Client struct {
+	httpClient *http.Client
+	BaseURL    string
+}
+
+func New(app string, dialer agent.Dialer) *Client {
+	url := fmt.Sprintf("http://%s.internal:5500", app)
+
+	tr := &http.Transport{
+		DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+			return dialer.DialContext(ctx, network, addr)
+		},
+	}
+
+	client := &http.Client{Transport: tr}
+
+	return &Client{
+		httpClient: client,
+		BaseURL:    url,
+	}
+}
+
+func (c *Client) Do(ctx context.Context, method, path string, in, out interface{}) error {
+	req, err := c.NewRequest(path, method, in)
+	if err != nil {
+		return err
+	}
+
+	req = req.WithContext(ctx)
+
+	res, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.Status > "299" {
```

Comment:
What are the expected status codes?  Also, looks like you're comparing an `int` with a `string` here.

Reply:
True I'll work on it.

Reply:
Will we be working with any valid status codes, other than 200 and 201?